bin/check_components_diff.rb

```diff
@@ -33,18 +34,29 @@ def print_details(component, details)
     end
   end
 
-  def add_row(component, tag, changes_exist = false)
-    revision = tag.match(/[0-9]+\.[0-9]+\.[0-9]+/)
+  def add_row(component, revision, changes_exist = false)
     rows << if changes_exist
-              [component, revision.to_s.green, nil]
-            else
-              [component, revision.to_s.red, (change_revision revision.to_s).green]
+              [component, revision.green, nil]
+            else change_version(component, revision)
             end
   end
+end
 
-  def change_revision(revision)
-    revision.gsub(/[0-9]+$/) { |s| (s.to_i + 1).to_s }
+def change_version(component, revision)
+  if !ARGV[1] || ARGV[1] == 'z'
+    [component, revision.red, (revision.gsub(/[0-9]+$/) { |s| (s.to_i + 1).to_s }).green]
+  else
+    [component, revision.red, change_version_y(revision).yellow]
   end
 end
 
+def change_version_y(revision)
+  revision.gsub!(/\.[0-9]+\./) { |s| ".#{(s.delete('.').to_i + 1)}." }
+  revision.gsub(/[0-9]+$/, '0')
+end
```

Comment:
Should move these methods inside the class, and move the class in a folder `lib/presenters`